### PR TITLE
Use electron-builder and webpack@3

### PR DESCRIPTION
**Pull request by n-riesco**
@chriddyp Could you review this PR?

---

I haven't gone as far as I originally planned with this work, but I'm going to open a PR in the main repo, so that people can test the installers generated with `electron-builder`.

A brief overview of what this PR does:
- now falcon uses a patched version of `ibm_db` from my repo with @tarzzz 's fix (this is necessary for `electron-builder` to work)
- I've upgraded dependencies if no code change was required (e.g. now we are using webpack@3 and mysql2)
- I've removed unused dependencies, files and folders.
- when possible I've moved modules from dependencies to devDependencies

Now the procedure to generate installers is:
```
$ yarn install
$ yarn build
$ yarn run pack
```
(Beware that `yarn pack` and `yarn run pack` do different things)

The procedure to build the app locally and test is still the same:
```
$ yarn install
$ yarn run rebuild:modules:electron
$ yarn build
$ yarn start
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -156,10 +157,11 @@
     "fetch-cookie": "^0.6.0",
     "fixed-data-table": "^0.6.4",
     "form-data": "^2.3.1",
-    "fs-extra": "https://registry.npmjs.org/fs-extra/-/fs-extra-0.26.7.tgz",
+    "fs-extra": "^4.0.2",
+    "immutable": "^3.8.2",
```
Comment (chriddyp) [inline package.json]: I'm pretty sure that we've gotten rid of immutable everywhere, and that these prop type descriptions https://github.com/plotly/falcon-sql-client/blob/92755984257409dcdcfb29231e272d8e4d2d174d/app/components/Settings/Logger/Logger.react.js#L33 are actually incorrect. We can change this in a future PR though

Comment (n-riesco) [reply package.json]: I kept it to silence the warning that `immutable` is a peer depedency of `react-immutable-proptypes`.

Comment (chriddyp) [reply package.json]: right - i mean, we can probably remove all of the `react-immutable-proptypes` throughout our code as well

**Comment by chriddyp** [comment]
this looks great! 💃

**Pull request merged**